### ISSUE TITLE
ServiceWorker error for GoogleBot

### ISSUE THREAD

**Issue by davyvanlaere** [angular/angular#40292]
Not really sure if this is an issue: this errors occurs on the client when GoogleBot visits, and when ServiceWorkerModule.register is being called.

Not sure yet if this affects indexing, but i would assume GoogleBot does not care about js errors.

The output from my logging:

`{
	"value": "Error: Rejected\n    at ServiceWorkerContainer.wrsParams.serviceWorkers.navigator.serviceWorker.register (<anonymous>:10:852)\n    at n._next (https://www.free-sudoku-puzzle.net/main.417f5bc5ef2630cdc7a3.js:1:591150)\n    at n.value (https://www.free-sudoku-puzzle.net/main.417f5bc5ef2630cdc7a3.js:1:23649)\n    at n.value (https://www.free-sudoku-puzzle.net/main.417f5bc5ef2630cdc7a3.js:1:22788)\n    at n.value (https://www.free-sudoku-puzzle.net/main.417f5bc5ef2630cdc7a3.js:1:21676)\n    at n.value (https://www.free-sudoku-puzzle.net/main.417f5bc5ef2630cdc7a3.js:1:21306)\n    at n.value (https://www.free-sudoku-puzzle.net/main.417f5bc5ef2630cdc7a3.js:1:72357)\n    at n.value (https://www.free-sudoku-puzzle.net/main.417f5bc5ef2630cdc7a3.js:1:21306)\n    at n.value (https://www.free-sudoku-puzzle.net/main.417f5bc5ef2630cdc7a3.js:1:20214)\n    at n.value (https://www.free-sudoku-puzzle.net/main.417f5bc5ef2630cdc7a3.js:1:594348)",
	"ua": "Mozilla/5.0 AppleWebKit/537.36 (KHTML, like Gecko; compatible; Googlebot/2.1; +http://www.google.com/bot.html) Chrome/87.0.4280.90 Safari/537.36",
}`

The shortened code of my module:

`@NgModule({
  declarations: [
    AppComponent, NavigationComponent
  ],
  imports: [
    BrowserModule,
    SharedModule.forRoot(),
    HttpClientModule,
	...,
    ServiceWorkerModule.register('ngsw-worker.js', { enabled: environment.production })
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }`

**Comment by gkalpak** [comment]
Thx for raising this issue, @davyvanlaere!

I looked into it and I believe this working as intended. This behavior doesn't seem to be specific to Angular's SW. I tried on a project of mine with a hand-crafted SW (no framework) and got the same error. Also, [this user][1] reported the same error on a React project.

So, I assume that this is expected (and intended, since GoogleBot wouldn't want a SW in effect when navigating to a page) and thus does not affect indexing.

I am going to close the issue, because I don't think there is any action to take, but feel free to continue the discussion below. 

[1]: https://stackoverflow.com/questions/63301353/service-worker-registeration-failed-when-indexed-in-google-search-console
